### PR TITLE
Improved fish shell completions

### PR DESCRIPTION
**Pull request by mymmrac**
- Added check for non-zero exit code of `task --list-all` that happens if there is no `Taskfile` present or etc.
- Added comments
- Added fish to `.editorconfig`

### CODE REVIEW

File: completion/fish/task.fish [hunk 1]
```diff
@@ -27,3 +35,4 @@ complete -c $GO_TASK_PROGNAME -s t -l taskfile  -d 'choose which Taskfile to run
 complete -c $GO_TASK_PROGNAME -s v -l verbose   -d 'enables verbose mode'
 complete -c $GO_TASK_PROGNAME      -l version   -d 'show Task version'
 complete -c $GO_TASK_PROGNAME -s w -l watch     -d 'enables watch of the given task'
+
```
Comment (andreynering) [inline completion/fish/task.fish:38]: This extra empty line here is not needed. (Should have a single empty line, we now have two).

**Review by andreynering — APPROVED** [review]
Hi @mymmrac, thanks for this PR!

I see you changed indentation from spaces to tabs. Can you use 2 space indention? That's what we've been using for all non-Go files.

I'd also appreciate if you added `fish` to the file extension list here:

https://github.com/go-task/task/blob/6335878317390ad5535d73d9898e91028d75b538/.editorconfig#L11

Squashing all changes and keeping a single commit is preferred.

File: completion/fish/task.fish (hunk 1 above)
Comment (mymmrac) [reply completion/fish/task.fish:38]: Hmm, previously there were no new line at the end, so I added it, right now it looks like this:
image

Comment (mymmrac) [reply completion/fish/task.fish]: Removed that to match all other files

**Comment by andreynering** [comment]
Perfect, thanks @mymmrac!

**Pull request merged**